edge/pkg/edgestream/edgestream.go

```diff
@@ -73,27 +73,32 @@ func (e *edgestream) Start() {
 		Path:   "/v1/kubeedge/connect",
 	}
 	// TODO: Will improve in the future
-	ok := <-edgehub.HasTLSTunnelCerts
-	if ok {
-		cert, err := tls.LoadX509KeyPair(config.Config.TLSTunnelCertFile, config.Config.TLSTunnelPrivateKeyFile)
-		if err != nil {
-			klog.Exitf("Failed to load x509 key pair: %v", err)
-		}
-		tlsConfig := &tls.Config{
-			InsecureSkipVerify: true,
-			Certificates:       []tls.Certificate{cert},
-		}
+	if ok := <-edgehub.HasTLSTunnelCerts; !ok {
+		klog.Exitf("Failed to write cert key pair into disk")
+	}
 
-		for range time.NewTicker(time.Second * 2).C {
-			select {
-			case <-beehiveContext.Done():
-				return
-			default:
-			}
+	cert, err := tls.LoadX509KeyPair(config.Config.TLSTunnelCertFile, config.Config.TLSTunnelPrivateKeyFile)
+	if err != nil {
+		klog.Exitf("Failed to load x509 key pair: %v", err)
+	}
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
+		Certificates:       []tls.Certificate{cert},
+	}
+
+	ticker := time.NewTicker(time.Second * 2)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-beehiveContext.Done():
+			return
+		case <-ticker.C:
```

Comment:
`time.After` is it more comfortable?

Reply:
No, I don't think so. time.After does not work that way, it not a ticker.

Reply:
Every time you call time.After, it new a Timer object, ticker is the standard solution here.